### PR TITLE
use index for refactoring operations

### PR DESCRIPTION
**Pull request by nittka**
As described in #118 there are severe performance issues in the refactoring operations if the workspace contains many LilyPond source files. This is because in the prior version all the includes of these files where re-evaluated.

I added include information to the index which allows to calculate which source files are affected by a change much faster. At the same time, I changed the refactoring process in a way such that moving multiple files and folders in one refactoring works (fix for #52).

I added preference switches that allow to suppress refactoring warnings that a file contains variable includes (which cannot be refactored as it is unclear which variable is used) as well as the warning that a configured search path is affected by the refactoring (i.e. the search path points to a folder in the workspace and this folder is part of the refactoring which should be a rare case).

I tried to make the refactoring deal with as many cases as possible (relative and absolute includes within the workspace, relative and absolute includes pointing outside the workspace, includes using a search path inside and outside the workspace).

The prior (?) and the current approach have limitation (I'd say severity critical for a case that never occurs). Files outside the workspace are not considered in the refactoring: X is the file containing includes and T is the refactoring target (e.g. a file that is renamed). If there is an include X->T, the reference in X is indexed and updated and everything is fine. If there is an include X->A->T, where A is a file outside the workspace, A is not indexed and hence its reference to T not updated.

### CODE REVIEW

**Comment by nittka** [comment]
The initial refactoring caused the unit tests to fail (as now I am checking whether absolute includes are actually within the ws, but in the unit tests there is no workspace) ^-^
Commit 7d8cacfb9f044b21414cf6587053fed074f12e17 fixes this and also adds the regression test for #86, so the corresponding branch is now obsolete.

File: org.elysium.test/src/org/elysium/test/regression/GrammarRegressions.xtend [hunk 1]
```diff
@@ -370,7 +370,7 @@ class GrammarRegressions {
 
 	@Test
 	def void hyphenAfterCommand() {
-		//TODO is f the command for f-1? 
+		//TODO is f the command or f-1?
```
Comment (thSoft) [inline org.elysium.test/src/org/elysium/test/regression/GrammarRegressions.xtend]: \f is the command. See http://lilybin.com/8uaqk5/1

File: org.elysium.ui/plugin.xml [hunk 2]
```diff
@@ -375,6 +377,22 @@
         <command commandId="org.elysium.ui.commands.RecompileSelected"/>
       </menu>
     </menuContribution>
+    <menuContribution locationURI="popup:org.elysium.ui.views.ScoreLibrary?after=group.goto">
+      <menu commandId="org.eclipse.ui.navigate.showInQuickMenu"
+          id="org.elysium.ui.showInMenu"
+          label="Show In">
+        <!--dynamic
```
Comment (thSoft) [inline org.elysium.ui/plugin.xml]: Please delete commented out XML.

**Comment by thSoft** [comment]
Great! Do you plan to push further commits to this PR? If yes, I'd rather cherry-pick 16f4d95 because it addresses a separate issue.

**Comment by nittka** [comment]
No, no further features, just changes according to your comments... I guess except for the initial commit and 7d8cacfb9f044b21414cf6587053fed074f12e17, they all address different issues ;-)

File: org.elysium/src/org/elysium/GenerateLilyPond.mwe2 [hunk 3]
```diff
@@ -57,7 +57,7 @@ Workflow {
 			}
 
 			fragment = validation.JavaValidatorFragment {
-				composedCheck = "org.eclipse.xtext.validation.ImportUriValidator"
+//				composedCheck = "org.eclipse.xtext.validation.ImportUriValidator"
```
Comment (thSoft) [inline org.elysium/src/org/elysium/GenerateLilyPond.mwe2]: My usual comment applies about commented out code. ;)

**Comment by nittka** [comment]
It's not that I wanted to ignore your last comment about removing the comment from the LilyPond.mwe2 file - I was expecting further comments before you merged the pull requests...

**Pull request merged**